farm/infer.py

```diff
@@ -189,42 +178,50 @@ def inference_from_dicts(self, dicts, rest_api_schema=False, use_multiprocessing
 
                 preds_all = []
                 with tqdm(total=len(dicts), unit=" Dicts") as pbar:
-                    for dataset, tensor_names, sample in results:
-                        preds_all.extend(self._run_inference(dataset, tensor_names, sample))
+                    for dataset, tensor_names, baskets in results:
+                        samples= []
+                        for b in baskets:  # number of baskets in _multiproc() related to chunksize
+                            samples.extend(b.samples)
+                        if self.prediction_type == "span_classification":
+                            preds_all.append(self._run_inference_qa(dataset, tensor_names, samples))
+                        else:
+                            preds_all.extend(self._run_inference(dataset, tensor_names, samples))
                         pbar.update(multiprocessing_chunk_size)
 
         else:
             chunk = next(grouper(dicts, len(dicts)))
-            dataset, tensor_names, sample = self._multiproc(chunk, processor=self.processor, rest_api_schema=rest_api_schema)
-            preds_all = self._run_inference(dataset, tensor_names, sample)
+            dataset, tensor_names, baskets = self._multiproc(chunk, processor=self.processor, rest_api_schema=rest_api_schema)
```

Comment:
Why not getting samples back from _multiproc() instead of baskets (as we did before)? Would make the code here easier to read by moving the unpacking part to multiproc()